pkg/controller/certificates/issuing/issuing_controller.go

```diff
@@ -279,25 +279,31 @@ func (c *controller) ProcessItem(ctx context.Context, key string) error {
 		return nil
 	}
 
-	// Some issuers won't honor the "Denied=True" condition, and we don't want
-	// to break these issuers. To avoid breaking these issuers, we skip bubbling
-	// up the "Denied=True" condition from the certificate request object to the
-	// certificate object when the issuer ignores the "Denied" state.
-	//
-	// To know whether or not an issuer ignores the "Denied" state, we pay
-	// attention to the "Ready" condition on the certificate request. If a
-	// certificate request is "Denied=True" and that the issuer still proceeds
-	// to adding the "Ready" condition (to either true or false), then we
-	// consider that this issuer has ignored the "Denied" state.
-	if crReadyCond == nil {
-		if apiutil.CertificateRequestIsDenied(req) {
-			return c.failIssueCertificate(ctx, log, crt, apiutil.GetCertificateRequestCondition(req, cmapi.CertificateRequestConditionDenied))
-		}
+	// Now check if CertificateRequest is in any of the final states so that
```

Comment:
Can we add a comment here that explains that we have to deny the CertificateRequest even if the  Ready condition is set to Issued because otherwise we have a race condition?

Reply:
I think we should only communicate the fact that we consider Denied condition to be final. I can add that it takes priority over Ready condition with Issued reason, hopefully that'll make it clearer. I think a comment that talks about a bug that we are fixing today will be just confusing to someone reading it few months from now

Reply:
I updated the comment (and actually added something about the race condition too)